src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -4262,7 +4266,17 @@ .FORWARDHELPCATEGORY Cmdlet
     }
     else
     {
-        $help | more
+        # Respect PAGER, use more on Windows, and use less on Linux
+        if (Test-Path env:PAGER) {
+            $pager,$moreArgs = $env:PAGER -split '\s+'
+            $moreCommand = (Get-Command -CommandType Application $pager | Select-Object -First 1).Definition
+        } elseif ($IsWindows) {
+            $moreCommand = (Get-Command -CommandType Application more | Select-Object -First 1).Definition
+        } else {
+            $moreCommand = (Get-Command -CommandType Application less | Select-Object -First 1).Definition
+        }
+
+        $help | & $moreCommand $moreArgs $moreArgs
```

Comment:
IIRC someone (@BrucePay) had initialized `$moreArgs` to an empty string above the preceding if/else statements to prevent a strict mode failure ($moreArgs not being defined when $env:PAGER is not defined).

Reply:
https://github.com/PowerShell/PowerShell/pull/6781

Reply:
What about
```c#
        $pager,$moreArgs = $env:PAGER -split '\s+'
        if (Test-Path env:PAGER) {
            $moreCommand = (Get-Command -CommandType Application $pager | Select-Object -First 1).Definition
        } elseif ($IsWindows) {
            $moreCommand = (Get-Command -CommandType Application more | Select-Object -First 1).Definition
        } else {
            $moreCommand = (Get-Command -CommandType Application less | Select-Object -First 1).Definition
        }

        $help | & $moreCommand $moreArgs $moreArgs
```

Reply:
You have `$moreArgs` listed twice after the invocation of `$moreCommand`.  That should only be one.

Reply:
I updated my post.

Reply:
Also, watch out for `$pager != $null`.  That should be `-ne`.  That said, this line:
```
$pager,$moreArgs = $env:PAGER -split '\s+'
```
Will assign an empty string, **not** $Null, to $pager when `$env:PAGER` is not defined.  So I think the test should be simply: `if ($pager)`.   And ideally, if `get-command` can't find the app referred to by $env:PAGER, it should revert back to the default for the OS.  Otherwise, you left with a $null in $moreCommand and `help | & $null` will fail.

Reply:
Why are we doing `Get-Command -CommandType Application `? We get the same error for `badapp`, `& badapp` and `Get-Command -CommandType Application badapp`.

```c#
        $pager,$moreArgs = $env:PAGER -split '\s+'
        if ($pager) {
            $moreCommand = $pager
        } elseif ($IsWindows) {
            $moreCommand = "more.exe"
        } else {
            $moreCommand = "less"
        }

        $help | & $moreCommand $moreArgs

```